tilequeue/command.py

```diff
@@ -923,6 +924,60 @@ def tilequeue_enqueue_tiles_of_interest(cfg, peripherals):
     logger.info('%d tiles of interest processed' % n_toi)
 
 
+def coord_pyramids(coords, zoom_start, zoom_stop):
+    """
+    generate full pyramid for coords
+
+    Generate the full pyrmaid for the list of coords. Note that zoom_stop is
+    exclusive.
+    """
+    for coord in coords:
+        for child_coord in coord_children_range(coord, zoom_stop):
+            if zoom_start <= child_coord.zoom < zoom_stop:
+                yield child_coord
+
+
+def tilequeue_enqueue_full_pyramid_from_toi(cfg, peripherals, args):
+    """enqueue a full pyramid from the z10 toi"""
+    logger = make_logger(cfg, 'enqueue_tiles_of_interest')
+    logger.info('Enqueueing tiles of interest')
+
+    logger.info('Fetching tiles of interest ...')
+    tiles_of_interest = peripherals.toi.fetch_tiles_of_interest()
+    n_toi = len(tiles_of_interest)
+    logger.info('Fetching tiles of interest ... done')
+
+    zoom_start = args.zoom_start
+    zoom_stop = args.zoom_stop
+
+    rawr_yaml = cfg.yml.get('rawr')
+    assert rawr_yaml, 'Missing rawr yaml'
+    group_by_zoom = rawr_yaml.get('group-zoom')
+    assert group_by_zoom, 'Missing rawr group-zoom'
+    assert isinstance(group_by_zoom, int), 'Invalid rawr group-zoom'
+    assert zoom_start >= group_by_zoom
+
+    ungrouped = []
+    coords_at_group_zoom = set()
+    for coord_int in tiles_of_interest:
+        coord = coord_unmarshall_int(coord_int)
+        if coord.zoom < zoom_start:
+            ungrouped.append(coord)
+        if coord.zoom >= group_by_zoom:
```

Comment:
Since it has been asserted that `zoom_start >= group_by_zoom`, would it be possible to spell this as `else:`?

Reply:
`zoom_start` and `group_by_zoom` weren't the same, but [perhaps they should be](https://github.com/tilezen/tilequeue/pull/289#discussion_r149695722).

Reply:
Ah, right. I see now. It's possible for a `coord` to be below `zoom_start`, but above `group_by_zoom`, when `zoom_start` is set up to render those "in between" coords individually.

Sorry about that!